src/world/partner/watt.c

```diff
@@ -74,12 +91,125 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
     }
 }
 
+ApiStatus func_802BD2B4_31CE24(Evt *script, s32 isInitialCall) {
+    PlayerData* playerData = &gPlayerData;
+    Entity* entity;
+    f32 sp10, sp14;
+    f32 new_var;
+    f32 temp_f0;
+    Npc* npc = script->owner2.npc;
+    
+    if (gPartnerActionStatus.actionState.b[1] == 0) {
+        if (isInitialCall != 0) {
+            partner_flying_enable(npc, 1);
+            mem_clear(D_802BE274_31DDE4, sizeof(unk_802BE274_31DDE4));
+            D_8010C954 = NULL;
+        }
+
+        entity = D_8010C954;
+        playerData->unk_2F4[6]++;
+
+        if (entity == NULL) {
+            func_802BD100_31CC70(npc);
+            partner_flying_update_player_tracking(npc);
+            partner_flying_update_motion(npc);
+            if (npc->moveSpeed != 0.0f) {
+                if (D_802BE278_31DDE8 == 0) {
+                    D_802BE278_31DDE8 = 1;
+                    func_802BD1AC_31CD1C(1);
+                    npc->currentAnim.w = 0x60003;
+                }
+            } else if (D_802BE278_31DDE8 != 0) {
+                D_802BE278_31DDE8 = 0;
+                func_802BD1AC_31CD1C(0);
+                npc->currentAnim.w = 0x60001;
+            }
+
+            if (D_802BE310 != 0) {
+                D_802BE310->unk_0C->unk_04 = npc->pos.x;
+                D_802BE310->unk_0C->unk_08 = npc->pos.y + 13.0f;
+                D_802BE310->unk_0C->unk_0C = npc->pos.z;
+            }
+
+            return 0;
+
+        } else {
```

Comment:
this doesn't need to be an else if we're returning right before it. I think I like the return, so you can remove this else